### DISCUSSION TITLE
What is the right way to check programatically that the kwok cluster is ready in a test?

### DISCUSSION THREAD

**Discussion by reetasingh** [kubernetes-sigs/kwok#531] [General]
As part of https://github.com/kubernetes-sigs/e2e-framework/issues/214 , we are adding support for kwok cluster to e2e framework. I am curiuos on what is the right way to check kwok cluster is ready in a test? currently in this PR https://github.com/kubernetes-sigs/e2e-framework/pull/239/files I am checking if all 4 namespaces ("default", "kube-node-lease", "kube-public", "kube-system") are present. Is this the right way? what approach would you suggest

**Comment by wzshiming** [comment]
I think the right way is with the `--wait=` argument. 

``` console
$ kwokctl create cluster --help | grep "\--wait"
      --wait duration                           Wait for the cluster to be ready
```
